chiselFrontend/src/main/scala/chisel3/Bits.scala

```diff
@@ -855,6 +855,49 @@ sealed class UInt private[chisel3] (width: Width) extends Bits(width) with Num[U
   override def do_>> (that: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): UInt =
     binop(sourceInfo, UInt(this.width), DynamicShiftRightOp, that)
 
+  /**
+   * Circular shift to the left
+   * @param that number of bits to rotate
+   * @return UInt of same width rotated left n bits
+   */
+  final def rotateLeft(that: Int): UInt =  macro SourceInfoWhiteboxTransform.thatArg
+
+  def do_rotateLeft(n: Int)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): UInt = width match {
+    case KnownWidth(w) if n >= w => do_rotateLeft(n % w)
+    case _ if (n == 0) => this
```

Comment:
I think you need to make this `if (n == 0 || n == 1)`, or else the code for the n=1 case might either break (in presence of zero-width wires) or produce the wrong width of 2, not sure which.

Reply:
Do you mean `if (n == 0 || w == 0)`?

Reply:
Sorry,  I meant `n == 0 || w == 0`, yes.

Reply:
Yes a great catch!

Reply:
Should be fixed now.